### PR TITLE
amp-accordion implementation

### PR DESCRIPTION
**Pull request by dvoytenko**

### CODE REVIEW

File: src/slot.js [hunk 1]
```diff
@@ -72,18 +72,21 @@ export function Slot(props) {
...
+          } else {
+            // TBD: switch to explicit definition of "supported" attributes.
             node.setAttribute(name, value);
           }
         }
         // Boolean attributes:
         node.disabled = slot.hasAttribute('disabled');
         node.hidden = slot.hasAttribute('hidden');
         toggleAttribute(node, 'selected', slot.hasAttribute('selected'));
+        toggleAttribute(node, 'expanded', slot.hasAttribute('expanded'));
```
Comment (jridgewell) [inline src/slot.js]: This doesn't seem scalable to me. It seems like every AMP element will have to register what attributes it cares so synchronized between the slot and the element.

File: src/amp-react-accordion.js [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+  const [expanded, setExpanded] = useState({});
+  return preact.createElement(
+    'div',
+    {
+      ...props,
+    },
+    preact.createElement(
+      AmpAccordionContext.Provider,
+      {
+        ...props,
```
Comment (jridgewell) [inline src/amp-react-accordion.js]:
```suggestion
        children: props.children,
```

File: src/amp-react-accordion.js [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+  template: true,
+  // TODO: simply override ReactBaseElement instead.
+  init(reactBaseElement) {
+    const {element} = reactBaseElement;
+    const mu = new MutationObserver(records => {
+      rebuild();
+    });
+    const rebuild = () => {
+      const children = [];
+      const sectionChildren = element.querySelectorAll(':scope > section');
```
Comment (jridgewell) [inline src/amp-react-accordion.js]: Nit: I don't think `:scope` is necessary for this selector.

File: src/amp-react-accordion.js [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+      const postRender = () => {
+        // Skip mutations to avoid cycles.
+        mu.takeRecords();
+      };
+      sectionChildren.forEach((section, index) => {
```
Comment (jridgewell) [inline src/amp-react-accordion.js]: `map`?

File: src/slot.js (hunk 1 above)
Comment (dvoytenko) [reply src/slot.js]: I definitely agree. I think our `opts` system is definitely a way to go here. I'll do a separate PR to update all these cases. But also important to consider: there's definitely a small set of attributes that we would want to propagate to all slots. At the very least: `hidden` and `disabled`.

File: src/amp-react-accordion.js (hunk 2 above)
Comment (dvoytenko) [reply src/amp-react-accordion.js]: Done

File: src/amp-react-accordion.js (hunk 3 above)
Comment (dvoytenko) [reply src/amp-react-accordion.js]: There's no way to do a direct descendant selector without `:scope` in this case. Though in PROD code we may want to switch to just walking over `.children` given the `:scope` is not polyfillable.

File: src/amp-react-accordion.js (hunk 4 above)
Comment (dvoytenko) [reply src/amp-react-accordion.js]: Unfortunately it's a `NodeList` and it doesn't have a `map` method, only `forEach`. TBH I have no idea why they are messing with us - `map` would have been perfect and well-performing for `NodeList`s :(

**Pull request merged**